src/bplist.c

```diff
@@ -148,17 +171,31 @@ union plist_uint_ptr
 #define beNtoh(x,n) be64toh((x) << ((8-(n)) << 3))
 #endif
 
+#ifdef _MSC_VER
+static uint64_t UINT_TO_HOST(const void* x, uint8_t n)
+{
+		union plist_uint_ptr __up;
+		__up.src = (n > 8) ? (const char*)x + (n - 8) : x;
+		return (n >= 8 ? be64toh( get_unaligned_64(__up.u64ptr) ) :
+		(n == 4 ? be32toh( get_unaligned_32(__up.u32ptr) ) :
+		(n == 2 ? be16toh( get_unaligned_16(__up.u16ptr) ) :
+		(n == 1 ? *__up.u8ptr :
+		beNtoh( get_unaligned_64(__up.u64ptr), n)
+		))));
+}
+#else
 #define UINT_TO_HOST(x, n) \
 	({ \
 		union plist_uint_ptr __up; \
-		__up.src = ((n) > 8) ? (x) + ((n) - 8) : (x); \
+		__up.src = ((n) > 8) ? (const char*)(x) + ((n) - 8) : (x); \
```

Comment:
See comment below. I think we need another cast, like this:
```
__up.src = ((n) > 8) ? (const char*)(x) + ((n) - 8) : (const char*)(x);
```
And maybe in the _MSC_VER variant too.

Reply:
it does pass all checks though when I remove the cast you put there.